### PR TITLE
[NUI] Support markup anchor to TextLabel, TextField, TextEditor

### PR DESCRIPTION
**Pull request by wonrst**
### Description of Change ###

User can set href on anchors using the anchor tag <a> in markup.
and also can register AnchorTouched callback in Text components.
the AnchorTouched signal is emitted when the anchor is touced.

example:
label.Text = "\<a href='https://www.tizen.org'>Tizen<\/a>";
label.EnableMarkup = true;
label.AnchorTouched += (sender, e) =>
{
  Tizen.Log.Info("NUI", e.Href + "\n");
  Tizen.Log.Info("NUI", e.HrefLength + "\n");
};

This patch should be merged after below patches are applied.
https://review.tizen.org/gerrit/#/c/platform/core/uifw/dali-toolkit/+/253197/
https://review.tizen.org/gerrit/#/c/platform/core/uifw/dali-csharp-binder/+/253917/

### CODE REVIEW

File: src/Tizen.NUI/src/public/BaseComponents/TextEditorEvent.cs [hunk 1]
```diff
@@ -280,5 +334,66 @@ public TextEditor TextEditor
...
+                {
+                    href = value;
+                }
+            }
+            /// <summary>
+            /// Anchor href length.
+            /// </summary>
+            /// This will be public opened in tizen_6.5 after ACR done. Before ACR, need to be hidden as inhouse API.
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            public uint HrefLength
```
Comment (myroot) [inline src/Tizen.NUI/src/public/BaseComponents/TextEditorEvent.cs:386]: We don't need length, because string has length property

**Review by myroot — DISMISSED** [review]

File: src/Tizen.NUI/src/public/BaseComponents/TextFieldEvent.cs [hunk 2]
```diff
@@ -127,6 +166,21 @@ private void OnMaxLengthReached(IntPtr textField)
             }
         }
 
+        private void OnAnchorTouched(IntPtr textField, IntPtr href, uint hrefLength)
+        {
+            if (textFieldAnchorTouchedEventHandler != null)
```
Comment (myroot) [inline src/Tizen.NUI/src/public/BaseComponents/TextFieldEvent.cs:171]: Use null operator `?.`

``` c#
textFieldAnchorTouchedEventHandler?.Invoke(...)
```

**Comment by huiyueun** [comment]
Please add [NUI] tag to title.

File: src/Tizen.NUI/src/public/BaseComponents/TextEditorEvent.cs [hunk 3]
```diff
@@ -280,5 +334,66 @@ public TextEditor TextEditor
                 }
             }
         }
+
+        /// <summary>
+        /// The AnchorChanged event arguments.
+        /// </summary>
+        /// This will be public opened in tizen_6.5 after ACR done. Before ACR, need to be hidden as inhouse API.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public class AnchorTouchedEventArgs : EventArgs
```
Comment (huiyueun) [inline src/Tizen.NUI/src/public/BaseComponents/TextEditorEvent.cs:343]: Please check warning message.

File: src/Tizen.NUI/src/internal/Common/TextLabelSignal.cs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+namespace Tizen.NUI
+{
+    internal class TextLabelSignal : Disposable
+    {
+
+        internal TextLabelSignal(global::System.IntPtr cPtr, bool cMemoryOwn) : base(cPtr, cMemoryOwn)
+        {
+        }
+
```
Comment (huiyueun) [inline src/Tizen.NUI/src/internal/Common/TextLabelSignal.cs:28]: line 28 / line 34 -> delete 1 line.

File: src/Tizen.NUI/src/public/BaseComponents/TextEditorEvent.cs (hunk 1 above)
Comment (wonrst) [reply src/Tizen.NUI/src/public/BaseComponents/TextEditorEvent.cs]: I agree. It seems there is no need to provide HrefLength property in NUI.

File: src/Tizen.NUI/src/public/BaseComponents/TextFieldEvent.cs (hunk 2 above)
Comment (wonrst) [reply src/Tizen.NUI/src/public/BaseComponents/TextFieldEvent.cs]: Done

File: src/Tizen.NUI/src/public/BaseComponents/TextEditorEvent.cs (hunk 3 above)
Comment (wonrst) [reply src/Tizen.NUI/src/public/BaseComponents/TextEditorEvent.cs]: AnchorTouchedEventArgs is commonly used in TextLabel, TextField, and TextEditor.
So I placed this in a new file TextEvent.cs.
I think, later we can implement common EventArgs for Text components like TextChanged here.

**Review by huiyueun — APPROVED** [review]
LGTM

**Review by jaehyun0cho — APPROVED** [review]
LGTM

**Pull request merged**